src/io/flutter/devtools/DevToolsManager.java

```diff
@@ -56,18 +59,15 @@ public boolean hasInstalledDevTools() {
   }
 
   public CompletableFuture<Boolean> installDevTools() {
-    final FlutterSdk sdk = FlutterSdk.getFlutterSdk(project);
+    final FlutterSdk sdk =
+      FlutterSettings.getInstance().shouldUseBazel() ? FlutterSdk.forBazel(project) : FlutterSdk.getFlutterSdk(project);
```

Comment:
This expression appears twice:

```
FlutterSettings.getInstance().shouldUseBazel() ? FlutterSdk.forBazel(project) : FlutterSdk.getFlutterSdk(project)
```

Should we instead in-line this into `FlutterSdk`? I'm wondering if it's going to be rare or common. `FlutterSdk.getFlutterOrBazelSdk(project)`? I'm not sure there's an easy name for the function.

Reply:
I can pull this out into a separate method, however, we need to only use the bazelsdk in places where we are sure we want to: without changes to the internal bazel tooling, the only command that will work right now is the packages pub command.

Reply:
Gotcha. Having duplicates make sense if there are only a few (in this case 2), and if we do want to be explicit about what we're doing. It sounds like in this case we do - that a bazel flutter sdk is not a general replacement for a flutter sdk.